app/Http/Controllers/Dashboard/RolesController.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace App\Http\Controllers\Dashboard;
+
+use App\Models\Role;
+use App\Models\Permission;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Validator;
+use App\Http\Controllers\Controller;
+use Inertia\Inertia;
+use Inertia\Response;
+class RolesController extends Controller
+{
+    public function index(): Response
+    {
+         /**
+     * Shows the role page
+     * @return Response
+     */
+
+        return Inertia::render('Roles',
+            [
+                'roles' => []
+            ]
+        );
+    }
+    public function store(Request $request)
+    {
+        // Validate the request input
+        $validator = Validator::make($request->all(), [
+            'name' => 'required|string',
+            'display_name' => 'required|string',
+            'description' => 'required|string',
+        ]);
+
+        if ($validator->fails()) {
+            return response()->json($validator->errors(), 422);
```

Comment:
No need to do this. Inertia handles this if validation fails. 
You should consider using a request form here rather